apps/sms/js/thread_ui.js

```diff
@@ -1660,17 +1671,21 @@ var ThreadUI = {
     }
 
     // build messageDOM adding the links
-    messageDOM = this.buildMessageDOM(message, hidden);
-
-    messageDOM.dataset.timestamp = timestamp;
+    return this.buildMessageDOM(message, hidden).then((messageDOM) => {
+      messageDOM.dataset.timestamp = timestamp;
 
-    // Add to the right position
-    var messageContainer = this.getMessageContainer(timestamp, hidden);
-    this._insertTimestampedNodeToContainer(messageDOM, messageContainer);
+      // Add to the right position
+      var messageContainer = this.getMessageContainer(timestamp, hidden);
+      this._insertTimestampedNodeToContainer(messageDOM, messageContainer);
 
-    if (this.inEditMode) {
-      this.checkInputs();
-    }
+      if (this.inEditMode) {
+        this.checkInputs();
+      }
+      if (!hidden) {
```

Comment:
About this section: Seem not necessary to scroll while appendding a hidden message for first chunk.